src/app/Fake.Core.ReleaseNotes/ReleaseNotes.fs

```diff
@@ -66,16 +66,40 @@ type ReleaseNotes =
     static member New(assemblyVersion,nugetVersion,notes) = ReleaseNotes.New(assemblyVersion,nugetVersion,None,notes)
 
 let private parseVersions =
-    let nugetRegex = String.getRegEx @"([0-9]+.)+[0-9]+(-[a-zA-Z]+\d*)?(.[0-9]+)?"
+    // https://github.com/fsprojects/FAKE/issues/2557
+    let nugetRegexLegacy = String.getRegEx @"([0-9]+.)+[0-9]+(-[a-zA-Z]+\d*)?(.[0-9]+)?"
+    let nugetRegex = 
+        /// From Fake.Core.SemVer
+        let pattern =
```

Comment:
Can u please re-use patterns from `Fake.Core.SemVer` instead of duplicating them here?

Reply:
Can i pull `pattern` out of `(|SemVer|_|)` to use it? Like such?

```fsharp
let pattern =
      @"^(?<major>\d+)" +
      @"(\.(?<minor>\d+))?" +
      @"(\.(?<patch>\d+))?" +
      @"(\-(?<pre>[0-9A-Za-z\-\.]+))?" +
      @"(\+(?<build>[0-9A-Za-z\-\.]+))?$"

let (|SemVer|_|) version =
        match version with
        | ParseRegex pattern [major; minor; patch; pre; build] ->
            Some [major; minor; patch; pre; build]
        | _ ->
            None
```

Reply:
yeah sure